recipes/cpr/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CprConan(ConanFile):
+    name = "cpr"
+    description = "C++ Requests: Curl for People, a spiritual port of Python Requests"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://whoshuu.github.io/cpr/"
+    license = "MIT"
+    topics = ("conan", "cpr", "requests", "web", "curl")
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.with_openssl:
+            # If using OpenSSL, we need it to be active in libcurl too
+            if self.options["libcurl"].with_openssl == False:  # 'is False' doesn't work here (see conan-io/conan#3620)
+                raise ConanInvalidConfiguration("libcurl must be built with openssl support")
+            self.options["libcurl"].with_openssl = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("cpr-{}".format(self.version), self._source_subfolder)
+
+    def requirements(self):
+        self.requires("libcurl/7.67.0")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1d")
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["USE_SYSTEM_CURL"] = True
+        cmake.definitions["BUILD_CPR_TESTS"] = False
+        cmake.definitions["GENERATE_COVERAGE"] = False
+        cmake.definitions["USE_SYSTEM_GTEST"] = False
+        cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.with_openssl
+        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio" and self.options.shared:
+            cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
the minimum cmake version for this is 3.4 and you are setting 2.8. Please verify the original minimum version of the cmake provided by the library

Reply:
How should I do that? Using `cmake_minimum_required`? Checking for Windows + CMake version in `CMakeLists.txt`? Adding a check for the `cmake` version in the `conanfile.py`? Which one is preferred?

Reply:
The author's cmake requires 2.8.7, but there is a conditional for cmake 3.2. I would suggest changing the minimal version in the cmake wrapper to 3.4.3 due CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS

Reply:
Accepted suggestion by @uilianries in https://github.com/conan-io/conan-center-index/pull/450#pullrequestreview-336842461 👍